lib/chezmoi/boltpersistentstate.go

```diff
@@ -16,27 +16,26 @@ type BoltPersistentState struct {
 	db   *bolt.DB
 }
 
-// NewBoltPersistentState returns a new BoltPersistentState.
+// NewBoltPersistentState returns a new, unopened BoltPersistentState.
 func NewBoltPersistentState(fs vfs.FS, path string) (*BoltPersistentState, error) {
 	b := &BoltPersistentState{
 		fs:   fs,
 		path: path,
 		perm: 0600,
 	}
 	_, err := fs.Stat(b.path)
-	switch {
-	case err == nil:
-		if err := b.openDB(); err != nil {
-			return nil, err
-		}
-	case os.IsNotExist(err):
-	default:
+	if err != nil && os.IsNotExist(err) {
```

Comment:
I think my previous comment was incorrect. This should be:

```go
if err != nil && !os.IsNotExist(err) {
```

The logic should be: "either the stat should succeed or the file doesn't exist, anything else is an error".

Reply:
That makes more sense, yes.